lib/JSON/Validator/OpenAPI.pm

```diff
@@ -73,6 +73,9 @@ sub validate_request {
     }
 
     if ($type and defined($value //= $p->{default})) {
+      if ($type ne 'array' and ref $value eq 'ARRAY') {
```

Comment:
All query/formData/header data is returned as an array-ref, just like you implemented in the original PR @dboehmer.